catalog/api/app.py

```diff
@@ -0,0 +1,708 @@
+        if exception.body:
+            return web.Response(
+                body=exception.body,
+                headers={"Content-Type": "application/json"},
+                status=exception.status,
+            )
+        else:
+            return web.Response(
+                status=exception.status,
+            )
+    finally:
+        await api_client.close()
+
+@routes.get("/api/ratings/request/{request_uid}")
+async def provision_rating_get(request):
+    request_uid = request.match_info.get('request_uid')
+    user = await get_proxy_user(request)
+    email = user['metadata']['name']
+    return await api_proxy(
+        headers=request.headers,
+        method="GET",
+        url=f"{ratings_api}/api/ratings/v1/request/{request_uid}/email/{email}",
+    )
+
+@routes.post("/api/ratings/request/{request_uid}")
+async def provision_rating_post(request):
+    request_uid = request.match_info.get('request_uid')
+    user = await get_proxy_user(request)
+    data = await request.json()
+    data["email"] = user['metadata']['name']
+    return await api_proxy(
+        data=json.dumps(data),
+        headers=request.headers,
+        method="POST",
+        url=f"{ratings_api}/api/ratings/v1/request/{request_uid}",
+    )
+
+@routes.get("/api/ratings/catalogitem/{asset_uuid}/history")
+async def provision_rating_get(request):
+    asset_uid = asset.match_info.get('asset_uid')
+    user = await get_proxy_user(request)
+    session = await get_user_session(request, user)
+    if not session.get('admin'):
+        raise web.HTTPForbidden()
+    return await api_proxy(
+        headers=request.headers,
+        method="GET",
+        url=f"{ratings_api}/api/ratings/v1/catalogitem/{asset_uuid}/history",
+    )
+
+@routes.get("/api/admin/incidents")
+async def incidents_get(request):
+    return await api_proxy(
+        headers=request.headers,
+        method="GET",
+        params=request.query,
+        url=f"{admin_api}/api/admin/v1/incidents",
+    )
+
+@routes.post("/api/admin/incidents")
+async def create_incident(request):
+    user = await get_proxy_user(request)
+    session = await get_user_session(request, user)
+    if not session.get('admin'):
+        raise web.HTTPForbidden()
+    data = await request.json(),
+    return await api_proxy(
+        data=json.dumps(data),
+        headers=request.headers,
+        method="POST",
+        url=f"{admin_api}/api/admin/v1/incidents",
+    )
+
+@routes.post("/api/admin/incidents/{incident_id}")
+async def update_incident(request):
+    incident_id = request.match_info.get('incident_id')
+    user = await get_proxy_user(request)
+    session = await get_user_session(request, user)
+    if not session.get('admin'):
+        raise web.HTTPForbidden()
+    data = await request.json(),
+    return await api_proxy(
+        data=json.dumps(data),
+        headers=request.headers,
+        method="POST",
+        url=f"{admin_api}/api/admin/v1/incidents",
```

Comment:
I think this url should be 
url=f"{admin_api}/api/admin/v1/incidents/{incident_id}"

Reply:
Fixed.